### DISCUSSION TITLE
Getting metadata tags to print to a display

### DISCUSSION THREAD

**Discussion by Sparky54** [pschatzmann/arduino-audio-tools#743] [Q&A] [answered]
Hi, I’m working on an sd card audio player with an LCD display. I am largely using your library and everything has gone smoothly so far, except I haven’t been able to figure out how to read ID3V2 tags using the metadata functionality of the library to print information about the songs to the display. Any help or pointing in the right direction would be great. :)

**Comment by pschatzmann** [comment, marked as answer]
If you check in the class documentation you just need to [register a callback](https://pschatzmann.github.io/arduino-audio-tools/classaudio__tools_1_1_audio_player.html#ab32c4554cbd24248683bd0e5a0e1f360). This can also be a lambda function.

All of [player my examples](https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-player/player-sd-i2s/player-sd-i2s.ino) show how to do this.

**Reply by Sparky54** [reply]
how would I store values like the title in variables?

**Reply by pschatzmann** [reply]
Just define Arduino Strings as global variables and in the callback you can set (=copy) the values. The metadata type tells you what information you get...

**Reply by Sparky54** [reply]
how would I go about getting the bit rate from the decoder and the filesize from sdfat to calculate duration?

**Reply by pschatzmann** [reply]
This has nothing to do with the topic "Getting metadata": open a new discussion and do not forget to specify the relevant details. I have no clue what libraries you are using...
